EZP-27458: Aggregation API (#1171)
ezsystems/developer-documentation: docs/api/public_php_api_search.md

    Make sure that you set the Criterion on the `filter` property.
    It is impossible to use the `query` property, because the search in trash operation filters the database instead of querying.

## Aggregation

!!! caution "Feature support"

    Aggregation is only available in the Solr and Elasticsearch search engines.

> Aggregation enables you to find the count of search results for each Aggregation type.

Reviewer: ```suggestion
With aggregations you can find the count of search results for each Aggregation type.
```

> To do this, you need to make use of the query's `$aggregations` property:

Reviewer: ```suggestion
To do this, you use of the query's `$aggregations` property:
```

``` php
$query->aggregations[] = new ContentTypeTermAggregation('content_type');
```

> The name of the aggregation must be unique in the given query.

Reviewer: Is "content_type" the name of the aggregation? Could we use "<aggregation_name>" and explain that it can be, for example `content_type`?

Author: The whole Public PHP API doc section is based on actual examples that should be directly usable, the same name is used in examples further down.

Access the results by using the `get()` method of the aggregation:

``` php
$contentByType = $results->aggregations->get('content_type');
```

Aggregation results contain the name of the result and the count of found items:

``` php
foreach ($contentByType as $contentType => $count) {
    $output->writeln($contentType->getName() . ': ' . $count);
}
```

> Field aggregations enable you to group search results according to the value of a specific Field.

Reviewer: ```suggestion
With field aggregations you can group search results according to the value of a specific Field.
```
"According to" -> "based on"/"by"?

In this case the aggregation takes the Content Type identifier and the Field identifier as parameters.

The following example creates an aggregation named `selection` that groups results

> according to the value of the `topic` Field in the `article` Content Type:

Reviewer: Ditto

``` php
$query->aggregations[] = new SelectionTermAggregation('selection', 'article', 'topic');
```

> Tu use a range aggregation you additionally must provide a `ranges` array containing a set of `Range` objects

Reviewer: ```suggestion
To use a range aggregation, you must provide a `ranges` array containing a set of `Range` objects
```

that define the borders of the specific range sets.

``` php
$query->aggregations[] = new IntegerRangeAggregation('range', 'person', 'age',
[
    new Query\Aggregation\Range(1,30),
    new Query\Aggregation\Range(30,60),
    new Query\Aggregation\Range(60,null),
]);
```

!!! note

    The beginning of the range is included and the end is excluded,

>     so a range between 1 and 30 will include value `1`, but not include `30`.

Reviewer: ```suggestion
    so a range between 1 and 30 will include value `1`, but not `30`.
```
